### PR TITLE
54 implements posts findthreadsbycategoryid and findpostsbythreadid

### LINKED ISSUE #54: Implements posts findThreadsByCategoryId and findPostsByThreadId

### PR DESCRIPTION
**Pull request by reza-newgen**

### CODE REVIEW

File: src/main/java/edu/miu/cs545/project/service/impl/PostServiceImpl.java [hunk 1]
```diff
@@ -36,4 +40,40 @@ public Post create(Post post) {
...
+            Pageable pageable = PageRequest.of(page,size,sort);
+            Optional<ThreadPost> threadPostOpt = threadPostRepository.findById(id);
+            if(threadPostOpt.isPresent())
+                return postRepository.findByThreadPostAndParentPostIsNull(threadPostOpt.get(),pageable);
+
+        }catch (Exception e){
+        throw  new RuntimeException("Some thing happened in the server.");
+    }
+
+        return null;
```
Comment (darkhanzhursin) [inline src/main/java/edu/miu/cs545/project/service/impl/PostServiceImpl.java]: avoid returning null, instead use exception or throw

File: src/main/java/edu/miu/cs545/project/service/impl/PostServiceImpl.java [hunk 2]
```diff
@@ -36,4 +40,40 @@ public Post create(Post post) {
...
+            Pageable pageable = PageRequest.of(page,size,sort);
+            Optional<Post> postOpt = postRepository.findById(id);
+            if(postOpt.isPresent())
+                return postRepository.findByParentPost(postOpt.get(),pageable);
+
+        }catch (Exception e){
+            throw  new RuntimeException("Some thing happened in the server.");
+        }
+
+        return null;
```
Comment (darkhanzhursin) [inline src/main/java/edu/miu/cs545/project/service/impl/PostServiceImpl.java]: avoid returning null, instead use exception or throw

File: src/main/java/edu/miu/cs545/project/service/impl/ThreadPostServiceImpl.java [hunk 3]
```diff
@@ -1,13 +1,42 @@
...
+           Optional<ResourceCategory> categoryOpt = resourceCategoryRepo.findById(id);
+           Sort sort = Sort.by("id");
+            sort = sortDirection.equalsIgnoreCase("asc") ? sort.ascending() : sort.descending();
+            Pageable pageable = PageRequest.of(page,size,sort);
+           if(categoryOpt.isPresent())
+               return  threadPostRepository.findThreadPostByResourceCategory(categoryOpt.get(),pageable);
+        }catch (Exception e){
+            throw  new RuntimeException("Some thing happened in the server.");
+        }
+        return null;
```
Comment (darkhanzhursin) [inline src/main/java/edu/miu/cs545/project/service/impl/ThreadPostServiceImpl.java]: avoid returning null, instead use exception or throw

**Review by darkhanzhursin — CHANGES_REQUESTED** [review]
left comments

File: src/main/java/edu/miu/cs545/project/controller/PostController.java [hunk 4]
```diff
@@ -13,4 +17,28 @@ public class PostController extends CrudController<Post,Long>{
...
+
+    @Autowired
+    private PostService postService;
+
+    @GetMapping("/thread-post")
+    public Page<Post> getParentPostByThreadPost(@RequestParam(value = "id", required = true) Long id,
+                                                    @RequestParam(value = "page", required = false) Integer page ,
+                                                    @RequestParam(value = "size", required = false) Integer size,
+                                                    @RequestParam(defaultValue = "asc") String sortDirection){
+        if(null == page) page = 0;
```
Comment (nabil-miu) [inline src/main/java/edu/miu/cs545/project/controller/PostController.java]: let's make it if(page == null) same for if(size == null)

File: src/main/java/edu/miu/cs545/project/service/impl/PostServiceImpl.java [hunk 5]
```diff
@@ -36,4 +40,40 @@ public Post create(Post post) {
         }
         return postRepository.save(post);
     }
+
+
+    @Override
+    public Page<Post> findParentPostByThread(Long id, Integer page, Integer size, String sortDirection) {
+        try{
```
Comment (nabil-miu) [inline src/main/java/edu/miu/cs545/project/service/impl/PostServiceImpl.java]: formate please

**Review by nabil-miu — APPROVED** [review]

**Review by darkhanzhursin — APPROVED** [review]

**Pull request merged**